app/models/task.py

```diff
@@ -3,4 +3,29 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.id'))
+    goal = db.relationship("Goal", back_populates="tasks",
+        single_parent=True,
+        cascade="all, delete-orphan")
```

Comment:
I am not completely sure what role `cascade` would play here, but I suspect this is not needed.  Cascade & delete-orphan is usually used in a one to many relationship where removing the child from the parent should result in deleting the child (https://docs.sqlalchemy.org/en/14/orm/cascades.html#cascade-delete-orphan).  In this case, this is the 'many' side of the relationship, but even if the intention was to set it up on the `one` side, that doesn't quite fit the requirements, because removing a goal from a task shouldn't trigger deleting the task.